pontoon/base/views.py

```diff
@@ -295,6 +295,7 @@ def get_translations_from_other_locales(request):
 
     entity = get_object_or_404(Entity, pk=entity)
     locales = entity.resource.project.locales.exclude(code=locale)
```

Comment:
This is unrelated to your patch, but instead of using `locales` and `locale__in=locales`, we should use `locale = get_object_or_404(Locale, code=locale)` and `.exclude(locale=locale)` .

Reply:
Can you please clarify. If I eliminate `locales` and replace with `locale = get_object_or_404(Locale, code=locale)` I would only get the one locale that was passed into the method as opposed to all the locales in the project and then if I exclude with `locale=locale` it would produce an empty array.

Reply:
You also need to remove the `locale__in=locales` filter:
```python
translations = (
    Translation.objects
    .filter(
        entity=entity,
        plural_form=plural_form,
        approved=True,
    )
    .exclude(locale=locale)
    .order_by('locale__name')
)
```

Reply:
So, this works if I don't add `.exclude(locale=locale)`, but then it does not exclude the current locale. When I add the exclusion the app crashes and I get `undefined` as the value of `translations`

Also, while we are here....the list I am getting locally is not ordered by `locale__name` (this was the case even with the original code)

I have made some adjustments to the code in `entity.js` to parse the new data structure, which is is still a WIP but it seems to be parsing the lists correctly. I can push that if you would like to see as it's possible that is part of the problem.

Reply:
Nevermind.....I had a typo 😞

Reply:
However, it as I stated above, it's not sorting correctly for me and this was true even for the original code.  Which is why I did not have the `.order_by` on the `preferred_locales list`

Reply:
> Nevermind.....I had a typo 😞

Welcome to my World. :)

Feel free to push your WIP.

And yes - you should drop `.order_by('locale__name')`.